### PR TITLE
docs: fix update examples script

### LINKED ISSUE #1066: link to 404 in docs and readme
### Describe the bug

In the readme https://github.com/vitest-dev/vitest/tree/main/examples
In the docs https://vitest.dev/guide/#examples
the example link for testing-library-jest-dom points to a 404 error page
some other links might be broken

### Reproduction

https://github.com/vitest-dev/vitest/tree/main/examples

### System Info

```shell
chrome
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by userquin**
supersedes #1069

closes #1066

### CODE REVIEW

File: scripts/update-examples.ts [hunk 1]
```diff
@@ -3,19 +3,46 @@ import { promises as fs } from 'fs'
...
   const examplesRoot = resolve(fileURLToPath(import.meta.url), '../../examples')
 
-  const examples = await fs.readdir(examplesRoot)
-
   const data = await Promise.all(examples.map(async(name) => {
     const path = resolve(examplesRoot, name)
-    if ((await fs.lstat(path)).isFile())
+    const pkg = resolve(path, 'package.json')
+    try {
+      if (!(await fs.lstat(pkg)).isFile())
```
[the remark this answers is not in the document]
Comment (userquin) [reply scripts/update-examples.ts:40]: added error

[the remark this answers is not in the document]
Comment (userquin) [reply scripts/update-examples.ts:40]: uhhm, should we remove the catch on `lstat`?

**Review by patak-dev — APPROVED** [review]

**Comment by antfu** [comment]
Thanks. Fixed in cadefe56

**Comment by patak-dev** [comment]
@antfu how https://github.com/vitest-dev/vitest/commit/cadefe56868350b740e638b6559af37bccb6e2d2 fixes having an old example and running the scripts? 🤔
Most of the time there is a package.json in these local only examples, no? (I imagine added for testing or that didn't get deleted for some reason?)

**Pull request closed without merging**